allennlp/models/encoder_decoders/wikitables/wikitables_semantic_parser.py

```diff
@@ -136,12 +155,66 @@ def __init__(self,
                                                    mixture_feedforward=mixture_feedforward,
                                                    dropout=dropout)
 
+        self._executor_process: subprocess.Popen = None
+        self._create_sempre_executor()
+
+    def _create_sempre_executor(self) -> None:
+        """
+        Creates a server running SEMPRE that we can send logical forms to for evaluation.  This
+        uses inter-process communication, because SEMPRE is java code.  We also need to be careful
+        to clean up the process when our program exits.
+        """
+        if self._executor_process:
+            return
+        os.makedirs(SEMPRE_DIR, exist_ok=True)
+        abbreviations_path = os.path.join(SEMPRE_DIR, 'abbreviations.tsv')
+        if not os.path.exists(abbreviations_path):
+            subprocess.run(f'wget {ABBREVIATIONS_FILE}', shell=True)
+            subprocess.run(f'mv wikitables-abbreviations.tsv {abbreviations_path}', shell=True)
+
+        grammar_path = os.path.join(SEMPRE_DIR, 'grow.grammar')
+        if not os.path.exists(grammar_path):
+            subprocess.run(f'wget {GROW_FILE}', shell=True)
+            subprocess.run(f'mv wikitables-grow.grammar {grammar_path}', shell=True)
+
+        args = ['java', '-jar', cached_path(SEMPRE_EXECUTOR_JAR), 'serve', self._table_directory]
+        self._executor_process = subprocess.Popen(args,
+                                                  stdin=subprocess.PIPE,
+                                                  stdout=subprocess.PIPE,
+                                                  bufsize=1)
+
+        for _ in range(6):
+            # SEMPRE outputs six lines of stuff when it loads that I can't disable.  So, we clear
+            # that here.
+            self._executor_process.stdout.readline()
+        logger.info("Started SEMPRE server for evaluating logical forms")
+        atexit.register(self._stop_sempre_executor)
```

Comment:
What does this do? May be a comment would be useful here.

Reply:
Done.